### PR TITLE
feat(featureflags): get some metrics wired up

### PR DESCRIPTION
**Pull request by oliverb123**
Just a first step, then we can start actually emitting metrics for various things (DB reads and writes, and redis hits, in particular seem important). Most of the bulk of code change here is just fixing clippy lints in tests

Also changed some defaults to be more reasonable - theres going to be a lot of these pods and they won't be going through PgBouncer, we REALLY don't won't every one of them opening 100 connections.

### CODE REVIEW

File: rust/feature-flags/src/flag_matching.rs [hunk 1]
```diff
@@ -91,7 +91,7 @@ impl GroupTypeMappingCache {
                 return Err(e);
             }
         };
-        self.group_types_to_indexes = mapping.clone();
+        self.group_types_to_indexes.clone_from(&mapping);
```
Comment (dmarticus) [inline rust/feature-flags/src/flag_matching.rs:94]: I read up on this and it seems like `clone_from` is pretty generally a better operation.  Feels like I should replace this everywhere.  Thanks for bringing it up!

File: rust/feature-flags/src/router.rs [hunk 2]
```diff
@@ -26,9 +39,30 @@ where
...
-    Router::new()
+    let status_router = Router::new()
+        .route("/", get(index))
+        .route("/_readiness", get(index))
+        .route("/_liveness", get(move || ready(liveness.get_status())));
+
+    let flags_router = Router::new()
         .route("/flags", post(v0_endpoint::flags).get(v0_endpoint::flags))
-        .with_state(state)
+        .layer(ConcurrencyLimitLayer::new(concurrency))
```
Comment (dmarticus) [inline rust/feature-flags/src/router.rs:49]: Why add this?

File: rust/feature-flags/src/server.rs [hunk 3]
```diff
@@ -38,8 +41,23 @@ where
         }
     };
 
+    let health = HealthRegistry::new("liveness");
+
+    // TODO - we don't have a more complex health check yet, but we should add e.g. some around DB operations
+    let simple_loop = health
+        .register("simple_loop".to_string(), Duration::from_secs(30))
+        .await;
+    tokio::spawn(liveness_loop(simple_loop));
```
Comment (dmarticus) [inline rust/feature-flags/src/server.rs:50]: we have a TODO for adding a more complex one, but this is a great start for thinking about how to do it.

**Review by dmarticus — APPROVED** [review]
Thank you for doing this!  Seems like a good start so far :)

File: rust/feature-flags/src/database.rs [hunk 4]
```diff
@@ -5,20 +5,15 @@ use async_trait::async_trait;
...
 
-use crate::config::Config;
-
 const DATABASE_TIMEOUT_MILLISECS: u64 = 1000;
 
 #[derive(Error, Debug)]
 pub enum CustomDatabaseError {
-    #[error("Not found in database")]
-    NotFound,
-
```
[the remark this answers is not in the document]
Comment (oliverb123) [reply rust/feature-flags/src/database.rs:21]: Never shows up, and you're better off adding things are you encounter them than trying to guess ahead of time what you'll need as a general rule

File: rust/feature-flags/src/router.rs (hunk 2 above)
Comment (oliverb123) [reply rust/feature-flags/src/router.rs:49]: The config had a "concurrency" value, I figured the idea was to limit how many requests the pod is concurrently processing, and this is how you do that. Generally if requests _can_ result in significant memory usage, limiting how many you're processing at once is a good move.

**Pull request merged**